Sources/ETCD/ETCDClient.swift

```diff
@@ -120,5 +123,43 @@ public final class EtcdClient: @unchecked Sendable {
     public func put(_ key: String, value: String) async throws {
         try await put(key.utf8, value: value.utf8)
     }
+    
+    public func watch<Result>(_ key: some Sequence<UInt8>, _ operation: (WatchAsyncSequence) async throws -> Result) async throws -> Result {
+           let request = [Etcdserverpb_WatchRequest.with { $0.createRequest.key = Data(key) }]
+           let watchAsyncSequence = WatchAsyncSequence(watchClient.watch(request))
+           return try await operation(watchAsyncSequence)
+    }
+    
+    public func watch<Result>(_ key: String, _ operation: (WatchAsyncSequence) async throws -> Result) async throws -> Result {
+        let request = [Etcdserverpb_WatchRequest.with { $0.createRequest.key = Data(key.utf8) }]
+           let watchAsyncSequence = WatchAsyncSequence(watchClient.watch(request))
+           return try await operation(watchAsyncSequence)
+    }
+}
+
+public struct WatchAsyncSequence: AsyncSequence, AsyncIteratorProtocol {
+    public typealias Element = Data
+    let grpcAsyncSequence: GRPCAsyncResponseStream<Etcdserverpb_WatchResponse>
+    var grpcIterator: GRPCAsyncResponseStream<Etcdserverpb_WatchResponse>.AsyncIterator
+
+    init(_ grpcAsyncSequence: GRPCAsyncResponseStream<Etcdserverpb_WatchResponse>) {
+        self.grpcAsyncSequence = grpcAsyncSequence
+        self.grpcIterator = grpcAsyncSequence.makeAsyncIterator()
+    }
 
+    public func makeAsyncIterator() -> WatchAsyncSequence {
+        self
+    }
+    
+    public mutating func next() async -> Element? {
+        do {
+            guard let response = try await self.grpcIterator.next() else {
+                return nil
+            }
+            let element = try response.serializedData()
```

Comment:
Instead of mapping to `Data` here we have to map to this new structure `WatchEvent` that we have to create